### DISCUSSION TITLE
Flashing led using PWM

### DISCUSSION THREAD

**Discussion by marinehm** [micropython/micropython#11086] [ESP8266]
Learning to program n Micropython.  The best way I have learned is by using mini snippets like this and piecing them together to make things work.

Wemos ESP8266 Lite;
This script blinks the LED via PWM
PWM Frequency causing the LED to flash on/off
PWM Duty Cycle to adjust the brightness
This created an interesting flash pattern.

By: Nick Sebring
03-20-23

`from machine import Pin, PWM
from time import sleep

frequency = 5
led = PWM(Pin(5), frequency)

while True:
  for duty_cycle in range(0, 75):
    led.duty(duty_cycle)
    sleep(.0025)`

**Comment by bidrohini** [comment]
Thanks for sharing. I am doing micropython in Raspberry PI Pico. Maybe I will start with esp too someday.
